sklearn/model_selection/_validation.py

```diff
@@ -752,8 +750,9 @@ def learning_curve(estimator, X, y, groups=None,
     X, y, groups = indexable(X, y, groups)
 
     cv = check_cv(cv, y, classifier=is_classifier(estimator))
-    # Make a list since we will be iterating multiple times over the folds
+    # Store it as list as we will be iterating over the list multiple times
     cv_iter = list(cv.split(X, y, groups))
+
```

Comment:
I don't think we can do away with materializing for `learning_curve`. (Also it was like that before pre-0.18)